cmd/dataavailability/data_availability_check.go

```diff
@@ -229,8 +251,11 @@ func (d *DataAvailabilityCheck) checkDataAvailability(ctx context.Context, deliv
 	for url, reader := range d.urlToReaderMap {
 		_, err = reader.GetByHash(ctx, cert.DataHash)
 		if err != nil {
+			metrics.GetOrRegisterCounter(metricBase+"/"+url+"/failure", nil).Inc(1)
```

Comment:
You will need to canonicalize the metric names, see rpc_aggregator.go invalidPromCharRegex.

You might want to move that to a package under utils/

Reply:
Done.